### PR TITLE
Kubernetes version validation

### LINKED ISSUE #469: Add missing Kubernetes version validation
We must ensure Kubernetes version is set when either of the following options is configured:
- manifest URLs
- local manifests
- Helm charts

### PR DESCRIPTION
**Pull request by dbw7**
Closes #469

### CODE REVIEW

File: pkg/image/validation/kubernetes.go [hunk 1]
```diff
@@ -414,3 +415,35 @@ func validateHelmChartDuplicates(charts []image.HelmChart) string {
 
 	return ""
 }
+
+func validateAdditionalArtifacts(ctx *image.Context) []FailedValidation {
+	var failures []FailedValidation
+
+	if _, err := os.Stat(combustion.KubernetesManifestsPath(ctx)); !os.IsNotExist(err) {
```
Comment (atanasdinov) [inline pkg/image/validation/kubernetes.go:422]: Let's instead just `ReadDir` directly - this call doesn't make it simpler.

File: pkg/image/validation/kubernetes.go [hunk 2]
```diff
@@ -414,3 +415,35 @@ func validateHelmChartDuplicates(charts []image.HelmChart) string {
...
+		dirEntries, err := os.ReadDir(combustion.KubernetesManifestsPath(ctx))
+		if err != nil {
+			failures = append(failures, FailedValidation{
+				Error: fmt.Errorf("kubernetes manifests directory could not be read: %w", err),
```
Comment (atanasdinov) [inline pkg/image/validation/kubernetes.go:426]: You can use this description for the UserMessage and only set Error to the err value. Check its usages for reference.

File: pkg/image/validation/kubernetes_test.go [hunk 3]
```diff
@@ -14,7 +18,47 @@ var validNetwork = image.Network{
...
+	defer teardown()
+
+	valuesDir := filepath.Join(ctx.ImageConfigDir, "kubernetes", "helm", "values")
+	err := os.MkdirAll(valuesDir, os.ModePerm)
+	require.NoError(t, err)
+
+	apacheValuesPath := filepath.Join(valuesDir, "apache-values.yaml")
+	err = os.WriteFile(apacheValuesPath, []byte(""), 0o600)
+	require.NoError(t, err)
+
```
Comment (atanasdinov) [inline pkg/image/validation/kubernetes_test.go:61]: What's the point of all of this?

File: pkg/image/validation/kubernetes_test.go [hunk 4]
```diff
@@ -14,7 +18,47 @@ var validNetwork = image.Network{
 	APIVIP:  "127.0.0.1",
 }
 
+func setupContext(t *testing.T) (ctx *image.Context, teardown func()) {
```
Comment (atanasdinov) [inline pkg/image/validation/kubernetes_test.go:21]: We're barely using any of these directories - can't we just create it inline?

File: pkg/image/validation/kubernetes_test.go (hunk 3 above)
Comment (dbw7) [reply pkg/image/validation/kubernetes_test.go:61]: This is the only real way to test the Helm values file logic when it's working and in a non-error case

File: pkg/image/validation/kubernetes_test.go (hunk 4 above)
Comment (dbw7) [reply pkg/image/validation/kubernetes_test.go:21]: How about I greatly simplify the function?

**Review by atanasdinov — APPROVED** [review]

**Pull request merged**